File: resumeable_upload/golang/routes/route.go
package routes

import (
	"golang-note-resumeable-upload/controllers"
	"golang-note-resumeable-upload/services"

	"github.com/labstack/echo/v4"
)

func SetRoute(e *echo.Echo) {
	fileService := services.NewFileService()
	fileController := controllers.NewFileController(fileService)
	e.POST("/file/upload", fileController.Upload)
	e.POST("/file/merge", fileController.Merge)
	e.GET("/file/check-file/:fileId", fileController.CheckFile)
	e.POST("/file/upload-merge", fileController.UploadAndMerge)
}

File: resumeable_upload/rust/src/routes/route.rs
use std::sync::Arc;

use axum::{routing::{get, post}, Router};

use crate::{controllers::file_controller::{check_file_handler, merge_handler, upload_and_merge, upload_handler}, services::file_service::FileServiceImpl, states::app_state::AppState};

pub fn set_route() -> Router {
    let file_service = Arc::new(FileServiceImpl::new());
    let state = Arc::new(AppState{file_service: file_service.clone()});
    Router::new()
        .route("/file/upload", post(upload_handler))
        .route("/file/merge", post(merge_handler))
        .route("/file/check-file/{file_id}", get(check_file_handler))
        .route("/file/upload-merge", post(upload_and_merge))
        .with_state(state)
}